### PR TITLE
User authentication

### PR DESCRIPTION
**Pull request by sinfran**
- `authenticateUser` and `createUser` in `queries.js` handles login/sign up
- Sign up fails if email already exists in database 
- Login fails if email and password do not match or if email doesn't exist in database

### CODE REVIEW

**Review by danielanatolie — CHANGES_REQUESTED** [review]
You need to update what tables you're referencing since our sql script has been updated.

**Comment by sinfran** [comment]
@danielanatolie Fixed queries to match updated sql

File: client/src/App.js [hunk 1]
```diff
@@ -93,49 +103,127 @@ class App extends Component {
...
-                  }
-                />
-                <br />
-                <button 
-                type="submit">Submit</button>
-              </form>
-            </Modal>
-          ) : null}
-        </div>
+  <ReactBootstrap.Button type="button" onClick={this.handleShowModalClick}>
```
Comment (danielanatolie) [inline client/src/App.js]: Move the sign-in code into its own class under client/src/signin/signin.js refer to my current branch for an example of how the property table code is outside of App.js for easier debugging

**Review by danielanatolie — CHANGES_REQUESTED** [review]
For organization move the sign-in code

File: client/src/App.js (hunk 1 above)
Comment (sinfran) [reply client/src/App.js]: Ok -- currently working on it. I'm having a bit of trouble opening/closing the modal in a separate class, but I'll update the branch as soon as I get it working.

Comment (danielanatolie) [reply client/src/App.js]: App.js can control whether the modal is shown or not through the boolean 'showLoginModal'. You can pass in this boolean to your Login class from App.js and also pass in a function that sets it to false, here's a rough idea:

```
App.js
- boolean showLoginModal: true
- function hideLoginModal: this.setState({showLoginModal: false});
- function render(
      <Login showLoginModal hideLoginModal />
)

Login
- if (this.props.showLoginModal) 
      function render (
           ...bunch of cool stuff
            onClickSubmit -> this.props.hideLoginModal
      )
```

**Review by danielanatolie — CHANGES_REQUESTED** [review]
Resolve the conflicts, re-run to make sure it still works and then feel free to merge with master.

**Pull request merged**